src/EFCore.SqlServer/Query/Internal/SqlServerJsonPostprocessor.cs

```diff
@@ -49,6 +49,29 @@ public Expression Process(Expression expression)
         return Visit(expression);
     }
 
+    /// <summary>
+    ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+    ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+    ///     any release. You should only use it directly in your code with extreme caution and knowing that
+    ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+    /// </summary>
+    protected override Expression VisitExtension(Expression expression)
+        => expression switch
+        {
+            SqlServerOpenJsonExpression openJsonExpression
+                => openJsonExpression is { JsonExpression.TypeMapping: SqlServerStringTypeMapping { StoreType: "json" } } or
+                    { JsonExpression.TypeMapping: SqlServerJsonElementTypeMapping { StoreType: "json" } }
```

Comment:
nit:

```suggestion
                => openJsonExpression.JsonExpression.TypeMapping is
                    SqlServerStringTypeMapping { StoreType: "json" } or SqlServerJsonElementTypeMapping { StoreType: "json" }
```

Maybe add a small comment on which cases these two checks address (it's not obvious!)